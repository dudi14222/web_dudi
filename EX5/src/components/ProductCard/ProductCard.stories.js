import React from 'react';
import { storiesOf } from '@kadira/storybook';
import {
  withKnobs,
  text
} from '@kadira/storybook-addon-knobs';

import ProductCard from './';

const storiesProductCard = storiesOf('ProductCard', module);
storiesProductCard.addDecorator(withKnobs);

storiesProductCard.addWithInfo('with some props and children', '', () => {
  const story =  <ProductCard
  className="product-card" 
  style={{height: '480px',  backgroundImage: `url(${text('image url', 'https://www.lamborghini.com/en-en/sites/en-en/files/DAM/lamborghini/news/hightlight.jpg')})` }}
    >
    <p className="product-title">{text('title', 'title1')}</p>
  </ProductCard>;
 
  return story;
});


