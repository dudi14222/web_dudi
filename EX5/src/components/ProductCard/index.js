import React from 'react';

const ProductCard = ({ children, ...props }) => (
    <div className="col-md-3 col-sm-3">
        <div className="thumbnail" {...props}>
            {children}
        </div>
    </div>
)

export default ProductCard;
