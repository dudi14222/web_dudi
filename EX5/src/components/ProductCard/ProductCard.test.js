import ProductCard from '../ProductCard';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('ProductCard snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ProductCard />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('ProductCard shallow tests', () => {
    it('should render children when passed in', () => {
        const wrapper = shallow((
            <ProductCard>
                <div className="unique" />
            </ProductCard>
        ));    
        expect(wrapper.contains(<div className="unique" />)).toBe(true);
    });

    it('should set the props on thumbnail', () => {
        const wrapper = shallow((
            <ProductCard p1={"test1"} />               
        ));            
        expect(wrapper.find(".thumbnail").prop("p1")).toBe("test1");
    });
})



