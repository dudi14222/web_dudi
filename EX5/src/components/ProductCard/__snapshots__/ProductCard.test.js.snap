// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot tests test snapshot 1`] = `
<div
  className="col-md-3 col-sm-3"
>
  <div
    className="thumbnail"
  />
</div>
`;
